### ISSUE TITLE
Cannot lookup data for IDN domain

### ISSUE THREAD

**Issue by pnax** [zonemaster/zonemaster-gui#426]
Bug encountered while testing for v2023.1 (see https://github.com/zonemaster/zonemaster-gui/pull/410#issuecomment-1570614706).

Step to reproduce:
1. populate the input with an IDN domain such as `réussir-en.fr`
2. open the options
3. click on "Fetch NS from parent zone"
4. see the popup "No data found for the zone."

Expected behavior:
Data should be retrieved for the given zone.

I digged a little bit into this, and it looks like the Backend does not convert the input domain in `lookup_delegation_data`.

Should the client send a converted domain or should we let the Backend handle the conversion. Since the Backend can be used by multiple clients, I think the conversion should be done there. Maybe this issue should be moved in Zonemaster-Backend as well.

**Comment by matsduf** [comment]
It works to let the domain being tested through GUI. Does GUI do the conversion to A-label? When I try with `zmtest` it fails.
```
# zmtest réussir-en.fr
error: method job_create: {
  "code": "-32602",
  "data": [
    {
      "message": "The domain name is IDNA invalid",
      "path": "/domain"
    }
  ],
  "message": "Invalid method parameter(s)."
}
```

**Comment by blacksponge** [comment]
Fixed with 2023.2
